### PR TITLE
Feature/project structure update

### PR DESCRIPTION
**Pull request by fgulan**
With this PR we've added Carthage and SwiftPM support

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -379,37 +382,134 @@ return Single.just(loginModel)
...
+    targets: [
+        .target(
+            name: "YourDependency",
+            dependencies: [.product(name: "Japx", package: "Japx")]
+        )
+    ]
+)
+```
+
+We've also added some more functionalites like Alamofre or Moya for networking, Rx for reactive programming approach:
```
Comment (Dundo7) [inline README.md:463]: This is duplicated line from above in the Cocoapods example. 
Can we change it to something like:

If you want Alamofire or Moja for networking, Rx for reactive programming approach, here is how to add them/use them

File: README.md [hunk 2]
```diff
@@ -6,24 +6,28 @@
...
+  - [Usage](#usage)
     - [Codable](#codable)
     - [Codable and Alamofire](#codable-and-alamofire)
     - [Codable, Alamofire and RxSwift](#codable-alamofire-and-rxswift)
-- [Example project](#example-project)
-- [Authors](#authors)
-- [License](#license)
+  - [Installation](#installation)
+    - [Cocoapods](#cocoapods)
+    - [Swift Package Manager](#swift-package-manager)
```
[the remark this answers is not in the document]
Comment (Dundo7) [reply README.md:27]: Yes we should.

File: Examples/Carthage/Cartfile [hunk 3]
```diff
@@ -0,0 +1 @@
+github "Infinum/Japx" "feature/project-structure-update"
```
Comment (Dundo7) [inline Examples/Carthage/Cartfile:1]: Change this when we release the new version?

**Review by Dundo7 — APPROVED** [review]

File: Japx.podspec [hunk 4]
```diff
@@ -1,96 +1,56 @@
...
+  s.subspec "RxMoya" do |sp|
+    sp.source_files = "Japx/Classes/RxMoya/**/*"
+    sp.dependency "Japx/Moya"
+    sp.dependency "Moya/RxSwift"
   end
 
-  s.subspec 'ObjC' do |sp|
-      sp.source_files = 'Japx/Classes/ObjC/**/*'
-      sp.dependency 'Japx/Core'
+  s.subspec "ObjC" do |sp|
```
Comment (Truba) [inline Japx.podspec:52]: Why did you change everything to double quotes?
I was under the impression single quotes should be used in ruby as a default, and double only if we need string interpolation 🤔

**Review by Truba — APPROVED** [review]

File: Japx.podspec (hunk 4 above)
Comment (fgulan) [reply Japx.podspec:52]: Not sure to be honest, I didn't notice it - probably my ruby linter in VS code changed it to double quotes

**Comment by fgulan** [comment]
@Truba please check again

**Review by Truba — APPROVED** [review]
